source/slang/check.cpp

```diff
@@ -6945,25 +6945,16 @@ namespace Slang
 
         // Lookup generic parameter types in global scope
         for (auto name : entryPoint->genericParameterTypeNames)
-        {
-            firstDeclWithName = entryPoint->compileRequest->lookupGlobalDecl(name);
-            if (!firstDeclWithName)
-            {
-                // If there doesn't appear to be any such declaration, then
-                // we need to diagnose it as an error, and then bail out.
-                sink->diagnose(translationUnitSyntax, Diagnostics::entryPointTypeParameterNotFound, name);
-                return;
-            }
-            RefPtr<Type> type;
-            if (auto aggType = firstDeclWithName->As<AggTypeDecl>())
-            {
-                type = DeclRefType::Create(entryPoint->compileRequest->mSession, DeclRef<Decl>(aggType, nullptr));
-            }
-            else if (auto typeDefDecl = firstDeclWithName->As<TypeDefDecl>())
-            {
-                type = GetType(DeclRef<TypeDefDecl>(typeDefDecl, nullptr));
-            }
-            else
+        {   
+            // parse type name
+            RefPtr<Expr> typeExpr = entryPoint->compileRequest->parseTypeString(entryPoint->getTranslationUnit(), name);
+            SemanticsVisitor visitor(
```

Comment:
This is a nice cleanup, even for the case where we just have a simple identifier.